hash_test.go

```diff
@@ -0,0 +1,52 @@
+package bn256
+
+import (
+	"testing"
+
+	"bytes"
+	"math/rand"
+	"strconv"
+)
+
+func TestKnownHashes(t *testing.T) {
+	for i, mh := range marshaledHashes {
+		g := HashG1([]byte{byte(i)})
+		if !bytes.Equal(mh[:], g.Marshal()) {
+			t.Fatal("hash doesn't match a known value")
+		}
+	}
+}
+
+func TestHashG1Collision(t *testing.T) {
```

Comment:
This test is not meaningful. Any hash algorithm is designed in such a way that finding a collision be hard.

Reply:
It was meant just a simple check if sth is not seriously wrong. Removed.